### DISCUSSION TITLE
Proof of Compatibility lemma 3 in Logical Relations

### DISCUSSION THREAD

**Discussion by Bohun9** [ppolesiuk/type-systems-notes#28] [Q&A] [answered]
I thought that we will not use nameless representation for variables as it makes proofs more technical, but in the proof below we use both x and function for handling nameless representation ⇑.

![image](https://github.com/ppolesiuk/type-systems-notes/assets/79287762/74a707cd-344c-4730-8076-5b699ff57b3d)

**Comment by ppolesiuk** [comment, marked as answer]
I've already changed the notation to be more similar to the one used during the lecture. In general it uses standard representation with names, but some notations are borrowed from Indexed Families of Sets, like $γ^*$, $γ^{\Uparrow_x}$, or $v \triangleleft_x γ$. The last two notations uses variable $x$ because we use standard variable-binding representation, but these notations were never introduced.

By the way, I think it is great idea that arose during the last lecture, to work with standard representation of binders, but use notation borrowed from nameless representation to narrow the gap between informal reasoning and formalization. I think we should consider adding a new section in the Syntax chapter that describes such notations used across the notes (and during the lecture).
